openedx/core/djangoapps/schedules/management/commands/send_verified_upgrade_deadline_reminder.py

```diff
@@ -0,0 +1,117 @@
+from django.core.management.base import BaseCommand
+from django.test.utils import CaptureQueriesContext
+from django.db.models import Prefetch
+from django.conf import settings
+from django.core.urlresolvers import reverse
+from django.db import DEFAULT_DB_ALIAS, connections
+
+from openedx.core.djangoapps.schedules.models import Schedule
+from openedx.core.djangoapps.user_api.models import UserPreference
+
+from edx_ace.message import MessageType
+from edx_ace.recipient_resolver import RecipientResolver
+from edx_ace import ace
+from edx_ace.recipient import Recipient
+
+
+from course_modes.models import CourseMode, format_course_price
+from lms.djangoapps.experiments.utils import check_and_get_upgrade_link
+
+
+class VerifiedUpgradeDeadlineReminder(MessageType):
+    pass
+
+
+class VerifiedDeadlineResolver(RecipientResolver):
+    def __init__(self, target_deadline):
+        self.target_deadline = target_deadline
+
+    def send(self, msg_type):
+        for (user, language, context) in self.build_email_context():
+            msg = msg_type.personalize(
+                Recipient(
+                    user.username,
+                    user.email,
+                ),
+                language,
+                context
+            )
+            ace.send(msg)
+
+    def build_email_context(self):
+        schedules = Schedule.objects.select_related(
+            'enrollment__user__profile',
+            'enrollment__course',
+        ).prefetch_related(
+            Prefetch(
+                'enrollment__course__modes',
+                queryset=CourseMode.objects.filter(mode_slug=CourseMode.VERIFIED),
+                to_attr='verified_modes'
+            ),
+            Prefetch(
+                'enrollment__user__preferences',
+                queryset=UserPreference.objects.filter(key='time_zone'),
+                to_attr='tzprefs'
+            ),
+        ).filter(
+            upgrade_deadline__year=self.schedule_deadline.year,
+            upgrade_deadline__month=self.schedule_deadline.month,
+            upgrade_deadline__day=self.schedule_deadline.day,
+        )
+
+        if "read_replica" in settings.DATABASES:
+            schedules = schedules.using("read_replica")
+
+        for schedule in schedules:
+            enrollment = schedule.enrollment
+            user = enrollment.user
+
+            user_time_zone = tzutc()
+            for preference in user.tzprefs:
+                user_time_zone = gettz(preference.value)
+
+            course_id_str = str(enrollment.course_id)
+            course = enrollment.course
+
+            course_root = reverse('course_root', kwargs={'course_id': course_id_str})
+
+            def absolute_url(relative_path):
+                return u'{}{}'.format(settings.LMS_ROOT_URL, relative_path)
+
+            template_context = {
+                'user_full_name': user.profile.name,
+                'user_personal_address': user.profile.name if user.profile.name else user.username,
```

Comment:
Nit: is there a better name for this?  Doesn't Personal address imply a residential address?  How about `user_personal_name`?